File: test_driver/pages/login_page_model.dart
import 'package:flutter_driver/flutter_driver.dart';

class LoginPageModel {
  final FlutterDriver driver;

  final emailField = find.byValueKey("emailField");
  final passwordField = find.byValueKey("passwordField");
  final submitButton = find.byType('ElevatedButton');

  LoginPageModel({required this.driver});

  Future<String> getEmailValue() async {
    return driver.getText(emailField);
  }

  Future<String> getPasswordValue() async {
    return driver.getText(passwordField);
  }

  Future<void> submitForm() async {
    return driver.tap(submitButton);
  }
}

File: appium_test/src/pages/loginPageModel.js


const wdio = require('webdriverio');
const find = require('appium-flutter-finder');




export default class LoginPageModel {

    emailField = find.byValueKey('emailField');
    passwordField = find.byValueKey('passwordField');



    static async getEmailValue() {
        const driver = await wdio.remote(opts);
        return driver.getElementText(this.emailField);
    }

}